src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/PreBuildEventValueProviderTests.cs

```diff
@@ -0,0 +1,524 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.VS.Properties.InterceptedProjectProperties;
+using Microsoft.VisualStudio.ProjectSystem.VS.Utilities;
+using Xunit;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ProjectSystemTrait]
+    public class PreBuildEventValueProviderTests
```

Comment:
Since the tests are a mirror of the PostBuildEvent, all the comments, above, apply here as well